generate_chip.py

```diff
@@ -581,9 +585,75 @@ def add_footprint_variant(
         else:
             raise ValueError('Either gap or footprints must be set')
 
+        # Generate 3D models (for certain package types)
+        if package_type in ['RESC', 'CAPC']:
+            uuid_3d = uuid('pkg', full_name, '3d')
+            if generate_3d_models:
+                generate_3d(library, package_type, full_name, uuid_pkg, uuid_3d, config)
+            package.add_3d_model(Package3DModel(uuid_3d, Name(full_name)))
+            for footprint in package.footprints:
+                footprint.add_3d_model(Footprint3DModel(uuid_3d))
+
         package.serialize(path.join('out', library, category))
 
 
+def generate_3d(
+    library: str,
+    package_type: str,
+    full_name: str,
+    uuid_pkg: str,
+    uuid_3d: str,
+    config: ChipConfig,
+) -> None:
+    import cadquery as cq
+    from cadquery_helpers import StepAssembly
+
+    print(f'Generating pkg 3D model "{full_name}": {uuid_3d}')
+
+    length = config.body.length
+    width = config.body.width
+    height = config.body.height
+
+    fillet = min(width, height) * 0.05
+    edge = min(length * 0.15, 0.4)
```

Comment:
Actually this should be derived from the `gap` value, or is it "hardcoded" by intention? Although the `gap` value is mainly intended for the footprint, indirectly it comes from the lead length of the body.

```suggestion
    edge = (length - config.gap) / 2
```

Reply:
Ah, true, probably makes sense.